root/hist/hist/THistFillBenchmarks.cxx

```diff
@@ -16,69 +13,227 @@ static void FillArguments(benchmark::internal::Benchmark *b)
          b->Args({i, j, j / 2});
 }
 
+// Create array for x coordinates
+double * MakeXCoorVec(int nbOfDataPoints) 
+{
+   double * vec = new double[nbOfDataPoints];
+   for (int j = 0; j < nbOfDataPoints; ++j) {
+      double d = static_cast< double >(j);
+      vec[j] = d / nbOfDataPoints;
+   }
+   return vec;
+}
 
+// Create array for y coordinates
+double * MakeYCoorVec(int nbOfDataPoints) 
+{
+   double * vec = new double[nbOfDataPoints];
+   for (int j = 0; j < nbOfDataPoints; ++j) {
+      double d = static_cast< double >(j);
+      vec[j] = d / (nbOfDataPoints / 10);
+   }
+   return vec;
+}
+
+// Create array for weights
+double * MakeWeightVec(int nbOfDataPoints) 
+{
+   double * vec = new double[nbOfDataPoints];
+   for (int j = 0; j < nbOfDataPoints; ++j) {
+      double d = static_cast< double >(j);
+      vec[j] = d;
+   }
+   return vec;
+}
+
+// Arrays of coordinates and weights
+double * xCoords = MakeXCoorVec(3000000);
+double * yCoords = MakeYCoorVec(3000000);
+double * weights = MakeWeightVec(3000000);
+
+// Histograms with different bin configurations for Fill
+TH2D hist1{"hist1", "", 10, 0., 1., 10 / 2, 0., 10.};
```

Comment:
Can we test overflow (or underflow) as well? E.g. by starting all x axes at 0.1?